File: cyanBat/src/main/java/at/smiech/cyanbat/fragments/GameOptionFragment.java
package at.smiech.cyanbat.fragments;

import android.os.Bundle;
import android.preference.PreferenceFragment;

import at.smiech.cyanbat.R;

/**
 * This fragment displays the game options based on Preferences to the user. The user can view and change preferences here.
 *
 * @author Martin Smiech
 */
public class GameOptionFragment extends PreferenceFragment {
    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        /**
         * Using xml resources to describe the layout of this fragment/PreferenceScreens.
         */
        this.addPreferencesFromResource(R.xml.gameoptions);
    }
}

File: cyanBat/src/main/java/at/smiech/cyanbat/fragments/GameOptionFragment.kt
package at.smiech.cyanbat.fragments

import android.os.Bundle
import androidx.preference.PreferenceFragmentCompat

import at.smiech.cyanbat.R

/**
 * This fragment displays the game options based on Preferences to the user. The user can view and change preferences here.
 *
 * @author msmiech
 */
class GameOptionFragment : PreferenceFragmentCompat() {
    override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
        /**
         * Using xml resources to describe the layout of this fragment/PreferenceScreens.
         */
        setPreferencesFromResource(R.xml.gameoptions, rootKey)
    }
}
